src/main/scala/ai/privado/policyEngine/PolicyExecutor.scala

```diff
@@ -188,8 +189,14 @@ class PolicyExecutor(
         matchingSourceIds.filter(ruleCache.getRuleInfo(_).get.isSensitive == sourceFilters.isSensitive.get)
 
     if (sourceFilters.name.nonEmpty)
-      matchingSourceIds =
-        matchingSourceIds.filter(ruleCache.getRuleInfo(_).get.name.toLowerCase.contains(sourceFilters.name.toLowerCase))
+      val namePattern: Option[Regex] = sourceFilters.name match {
+        case ""   => None
+        case name => Some(s"(?i)\\Q$name\\E".r)
```

Comment:
We already have a check in `if` condition for nonEmpty, why do we need a match expression again here?

Reply:
Also instead of forming regex on our end, better to pick whatever regex is passed in policies